lib/template.js

```diff
@@ -105,9 +110,27 @@ template.questions = function(templateBody, overrides) {
       if ('MaxValue' in parameter) valid = valid && Number(input) <= parameter.MaxValue;
       if (parameter.AllowedPattern) valid = valid && (new RegExp(parameter.AllowedPattern)).test(input);
       if (parameter.Type === 'List<Number>') valid = valid && input.split(',').every(isNumeric);
+
       return valid;
     };
 
+    question.filter = function(input) {
+      var done = this.async();
+
+      if (overrides.kms && parameter.Description.indexOf('[secure]') > -1 && input.slice(0, 7) !== 'secure:') {
+        return kms.encrypt({
+          KeyId: overrides.kms,
+          Plaintext: input
+        }, function(err, encrypted) {
+          if (err || !encrypted) return done(err);
```

Comment:
The contract from the aws-sdk is that you're either going to have an `err` or and `encrypted`, and never both and never neither. To cover the code paths that you've got here you have to mock the aws-sdk to do something that doesn't happen -- plus you get this one weird path where both are undefined and you call `done()` as if nothing went wrong.